packages/js/libsql-client/src/http/http-driver.ts

```diff
@@ -69,6 +75,42 @@ export class HttpDriver implements Driver {
             throw new Error(`${response.status} ${response.statusText}`);
         }
     }
+
+    private async queryUrl(): Promise<URL> {
+        if (this.version === undefined) {
+            this.version = await this.getVersion();
+            return this.queryUrl();
+        }
+
+        // perform version check when when the version is known.
+        return this.url;
+    }
+
+    async getVersion(): Promise<Version> {
+        let versionUrl = new URL("/version", this.url);
+        let resp = await compatibleFetch(versionUrl, {
+            method: "GET",
+        });
+
+        if (resp.status == 200) {
+            const versionStr = await resp.text();
+            const matches = versionMatcher.exec(versionStr);
+            if (matches === null) {
+                throw new Error(`invalid version format from server: ${versionStr}`);
+            }
+
+            let versionNums = matches.slice(1).map(x => parseInt(x, 10));
+            return [versionNums[0], versionNums[1], versionNums[2]];
+        }
+
+        // handle pre-version route
+        if (resp.status == 404) {
+            // dummy version < to any other version.
+            return [0, 0, 0]
+        }
+
+        throw new Error(await resp.text());
```

Comment:
What is this error going to tell me, the caller? Will it be specific or actionable?  It's going to preempt the original request that triggered it, which might be confusing in that context.